app/script/telemetry/calling/CallLogger.js

```diff
@@ -0,0 +1,186 @@
+z.telemetry.calling.CallLog = [];
+z.telemetry.calling.CallLogger = class CallLogger extends z.util.Logger {
+  constructor(message, options) {
+    super(message, options);
+    this.message = message;
+    this.options = options;
+  }
+
+  static get CONFIG() {
+    return {
+      IPV4: /(([0-1]?[0-9]{1,2}\.)|(2[0-4][0-9]\.)|(25[0-5]\.)){3}(([0-1]?[0-9]{1,2})|(2[0-4][0-9])|(25[0-5]))/,
+      IPV6: /(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))/,
+      MESSAGE_LOG_LENGTH: 10000,
+      OBFUSCATION_TRUNCATE_TO: 2,
+    };
+  }
+
+  obfuscate(string) {
+    //return CryptoJS.SHA256(`${string}${this.currentWeek}`)
+    //  .toString()
+    return string.substr(0, CallLogger.CONFIG.OBFUSCATION_TRUNCATE_TO);
+  }
+
+  obfuscateIp(ip) {
+    if (CallLogger.CONFIG.IPV4.test(ip)) {
+      // IPv4
+      ip = ip.split('.');
+      ip[ip.length - 1] = '0';
+      return ip.join('.');
+    } else if (CallLogger.CONFIG.IPV6.test(ip)) {
+      // IPv6
+      ip = ip.split(':').slice(0, 3);
+      return [...ip, '0000', '0000', '0000', '0000'].join(':');
+    }
+
+    return 'Unknown';
+  }
+
+  /*fakeIpGenerator(seed, type) {
+    if (type === 4) {
+      // Extend the 4 bytes seed to 16 bytes using XOR encryption with static keys
+      const originalSeedLength = seed.length;
+      const keys = [21, 15, 7];
+      for (let i = 0; i < originalSeedLength; ++i) {
+        for (const key of keys) {
+          seed += String.fromCharCode(key ^ seed.charCodeAt(i));
+        }
+      }
+
+      // Generate a fake IP from that seed
+      const fakeIp = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+      for (let i = 0; i < seed.length; i++) {
+        fakeIp[i % fakeIp.length] += seed.charCodeAt(i);
+      }
+      return fakeIp
+        .map(v => {
+          return (v % 10).toString(10);
+        })
+        .join('')
+        .match(/.{1,3}/g)
+        .join('.');
+    } else if (type === 6) {
+      const fakeIp = CryptoJS.MD5(seed).toString();
+      return fakeIp.match(/.{1,4}/g).join(':');
+    }
+  }
```

Comment:
remove if unused

Reply:
It will be needed in the future. Should I uncomment it then?